samples/tutorial/BindingToCommandHandlers.md

```diff
@@ -0,0 +1,62 @@
+
+```cs --source-file ./src/Binding/HandlerBindingSample.cs --project ./src/Binding/Binding.csproj --region MultipleArgs 
+```
+
+## Booleans (flags)
+
+If `"true"` or `"false"` is passed for an option having a `bool` argument, it is parsed and bound as expected. But an option whose argument type is `bool` doesn't require an argument to be specified. The presence of the option token on the command line, with no argument following it, results in a value of `true`. You can see various examples here:
+
+```cs  --source-file ./src/Binding/HandlerBindingSample.cs --project ./src/Binding/Binding.csproj --region Bool 
+```
+
+## Enums
+
+You can bind `enum` types as well. The values are bound by name, and the binding is case insensitive:
+
+```cs --source-file ./src/Binding/HandlerBindingSample.cs --project ./src/Binding/Binding.csproj --region Enum 
+```
+
+## Arrays, lists, and other enumerable types 
+
+Arguments having various enumerable types can be bound. A number of common types implementing `IEnumerable` are supported. In the next example, try changing the type of the `--items` `Option`'s `Argument` property to `Argument<IEnumerable<string>>` or `Argument<List<string>>`. 
+
+```cs --source-file ./src/Binding/HandlerBindingSample.cs --project ./src/Binding/Binding.csproj --region Enumerables 
+```
+
+## File system types
+
+Since command line applications very often have to work with the file system, `FileInfo` and `DirectoryInfo` are clearly important for binding to support. Run the following code, then try changing the generic type argument to `DirectoryInfo` and running it again.
+
+```cs --source-file ./src/Binding/HandlerBindingSample.cs --project ./src/Binding/Binding.csproj --region FileSystemTypes 
+```
+
+## Anything with a string constructor
+
+But `FileInfo` and `DirectoryInfo` are not special cases. Any type having a constructor that takes a single string parameter can be bound in this way. Go back to the previous example and try using a `Uri` instead.
+
+## More complex types
+
+Binding also supports creating instances of more complex types than we used in the examples above. If you have a large number of options, this can be cleaner than adding more parameters to your handler. `System.CommandLine` has the default convention of binding `Option` arguments to either properties or constructor parameters by name. The name matching uses the same strategies that are used when matching parameters on a handler method. 
+
+In the next sample, the handler accepts an instance of `ComplexType`. Try removing its setters and uncommenting the constructor. Try adding properties, or changing the types or names of its properties.
+
+```cs --source-file ./src/Binding/HandlerBindingSample.cs --project ./src/Binding/Binding.csproj --region ComplexTypes 
+```
+
+## Dependency injection
```

Comment:
Assuming we aren't doing full DI, consider softening this title - or give an example of using the programmer's own type

Reply:
By "full DI" do you mean "from a container"? We can fully support passing someone's custom types today (which may or may not come from a DI container, we're not really opinionated about it), but it probably deserves a separate tutorial because configuring this has little to do with binding.

Reply:
Maybe skip in this tutorial. I think there are two topics: the automatically available types you can have just by asking, and the ability to add your own type.

Reply:
Agreed. Mainly I wanted to point out that the capability exists in this context.

Related: https://github.com/dotnet/command-line-api/pull/533

Reply:
I agree with softening the title here. Perhaps something more like "built-in parameter types"?

Reply:
"System.CommandLine types" ?

Reply:
Yea I like that.